alpha/apps/kaltura/lib/batch2/bcdl/kBusinessPreConvertDL.php

```diff
@@ -1552,7 +1546,32 @@ private static function decideSourceFlavorConvert($entryId, assetParams $sourceF
 		kJobsManager::addFlavorConvertJob(array($srcSyncKey), $sourceFlavorOutput, $originalFlavorAsset->getId(), $conversionProfileId, $mediaInfoId, $parentJob);
 		return false;
 	}
-
+	
+		/*
+		 * Save the original source asset in another asset, in order 
+		 * to prevent its liquidation by the inter-source asset.
+		 */
+	private static function saveOriginalSource($mediaInfo)
+	{
+		$sourceAsset = assetPeer::retrieveById($mediaInfo->getFlavorAssetId());
+		$copyFlavorParams = assetParamsPeer::retrieveBySystemName(self::SAVE_ORIGINAL_SOURCE_FLAVOR_PARAM_SYS_NAME);
+		if (!$copyFlavorParams)
+			throw new APIException(APIErrors::OBJECT_NOT_FOUND);
+		
+		$asset = $sourceAsset->copy();
+		$asset->setFlavorParamsId($copyFlavorParams->getId());
+		$asset->setFromAssetParams($copyFlavorParams);
+		$asset->setStatus(flavorAsset::ASSET_STATUS_READY);
+		$asset->setIsOriginal(0);
+		$asset->setTags($copyFlavorParams->getTags());
+		$asset->incrementVersion();
+		$asset->save();
+		kFileSyncUtils::createSyncFileLinkForKey($asset->getSyncKey(asset::FILE_SYNC_ASSET_SUB_TYPE_ASSET), $sourceAsset->getSyncKey(asset::FILE_SYNC_ASSET_SUB_TYPE_ASSET));
+		$origFileSync = kFileSyncUtils::getLocalFileSyncForKey($sourceAsset->getSyncKey(asset::FILE_SYNC_ASSET_SUB_TYPE_ASSET));
```

Comment:
The local File-Sync could also be a link, you better use kFileSyncUtils::resolve.